mysql deprecation: add more troubleshooting steps (#7510)
mattermost/docs: source/deploy/postgres-migration.rst

Some common errors that you may encounter during the execution of the ``pgloader`` command are:

Invalid Input Syntax for Type JSON
::::::::::::::::::::::::::::::::::

If you receive an error message similar to the following:

.. code-block:: text

  ERROR Database error 22P02: invalid input syntax for type json

That would mean that the data in the MySQL database is not in a valid JSON format. You can fix this issue by updating the data in the MySQL database to be in a valid JSON format. To find out which row is causing the issue, you can run the following query (the ``<table_name>`` and ``<column_name>`` should be replaced with the actual table and column names that should've been indicated in the ``pgloader`` output):

.. code-block:: sql

  SELECT * FROM <table_name> WHERE JSON_VALID(<column_name>) = 0;

You can find and update the data in the MySQL database to be in a valid JSON format with the query above. After updating the data, you can run the ``pgloader`` command again.

Failed to Find Column or Table
::::::::::::::::::::::::::::::

If you receive an error message similar to the following:

.. code-block:: text

   pgloader failed to find column

> That would mean that the column or table is missing in the Postgres database. You can fix this issue by checking whether you have created the correct version of Postgres schema. After re-creating the shcema, you can run the ``pgloader`` command again.

Reviewer: ```suggestion
The column or table is missing in the PostgreSQL database. You can fix this issue by checking whether you have created the correct version of Postgres schema. After re-creating the schema, you can run the ``pgloader`` command again.
```

> Fell Through ECASE Expression

Reviewer: ```suggestion
Fell through ECASE expression
```

:::::::::::::::::::::::::::::

If you receive an error message similar to the following:

.. code-block:: text

   ERROR mysql: 76 fell through ECASE expression.

It is a `known issue <https://github.com/dimitri/pgloader/issues/1183>`_ with pgloader. You can fix this issue by either compiling ``pgloader`` from source or simply avoid this by running ``pgloader`` with our docker image. See: :ref:`install pgloader <deploy/manual-postgres-migration:install pgloader>` for more information.  

.. note::

>   Also, there may be cases where ``pgloader`` continue to migrate remaining tables and skip one or more tables during migration. In such cases, it is recommended to identify issues with the table and fix them before running the ``pgloader`` command again with a clean database. It is possible to run the ``pgloader`` command with the ``--debug`` flag to get more information about the errors.

Reviewer: ```suggestion
  Also, there may be cases where pgloader continues to migrate remaining tables and skip one or more tables during migration. In such cases, we recommend identifying issues with the table and fixing them before running the ``pgloader`` command again with a clean database. It is possible to run the ``pgloader`` command with the ``--debug`` flag to get more information about the errors.
```
